RFC0018.md

```diff
@@ -0,0 +1,69 @@
+
+When considering registration ordering, it is important to remember that PMIx has an inherent ordering inside it. Specifically, PMIx calls handlers in the following order:
+
+* single status event handlers - i.e., handlers that were registered against a single specific status.
+
+* multi status event handlers - those registered against more than one specific status
+
+* default event handlers - those registered against no specific status
+
+The intent of this PR is to add the missing capability to support arbitrary ordering within these categories - i.e., one can direct that a single status event handler come before another single status handler, but not that a multi-event handler precede a single event handler. There are only two allowed exceptions - a user may request that a specified event handler be executed at the beginning or the end of the chain, regardless of type.
+
+Within these constraints, the following attributes are proposed:
+
+* PMIX\_EVENT\_HDLR\_FIRST - put this event handler at the beginning of the chain and keep it there. Note that only one event handler can be given this directive - any subsequent registrations that contain this attribute will return an error, as will requests to "register before" this handler. Requests to "prepend" handlers will cause the new handler to immediately follow the one designated to be first. As previously noted, this event handler can be of any type.
+
+* PMIX\_EVENT\_HDLR\_LAST - put this event handler at the end of the chain. Note that only one event handler can be given this directive - any subsequent registrations that contain this attribute will return an error, as will requests to "register after" this handler. Requests to "append" handlers will cause the new handler to immediately precede the one designated to be last. As previously noted, this event handler can be of any type.
+
+* PMIX\_EVENT\_HDLR\_BEFORE - put this event handler immediately before the one specified in the (char*) value. The named event handler must be in the same category (single, multi, or default) as the one being registered. An error will be returned if the named event handler is not found.
+
+* PMIX\_EVENT\_HDLR\_AFTER  - put this event handler immediately after the one specified in the (char*) value. The named event handler must be in the same category (single, multi, or default) as the one being registered. An error will be returned if the named event handler is not found.
+
+#### Event notification extensions
+Two changes are proposed in this area, although only one actually impacts the standards header file by adding the following attribute:
+
+* PMIX\_EVENT\_NO\_TERMINATION - directs that all subsequent steps in the event handler chain must not call for termination of the application. Any other operations are permitted.
```

Comment:
So basically what you are proposing is that we allow each handler to modify the collected array of results, as opposed to only adding to them? We would definitely have to define which attributes can be overwritten, but implementation would be simple and not require an API change, if that's what people want to do. Again, worth bringing up in the meeting.